RT_UPDATE/check_and_load_latest_wpp.R

```diff
@@ -0,0 +1,151 @@
+#' @importFrom remotes install_github
+#' @export
+
+# function
+check_and_load_latest_wpp <- function() {
+  
+  # ---------------------------------------------------------------------------- #
+  # Step 1: Get all wpp-like repositories from GitHub
+  repos     <- jsonlite::fromJSON("https://api.github.com/users/PPgp/repos?per_page=100")
+  wpp_repos <- repos$name[grepl("^wpp\\d{4}$", repos$name)]
+  # ---------------------------------------------------------------------------- #
+  
+  # ---------------------------------------------------------------------------- #
+  # Step 2: Extract years and determine the latest
+  wpp_years <- as.integer(sub("wpp", "", wpp_repos))
+  if(length(wpp_years) == 0) {
+    
+    stop("No wpp packages found on GitHub.")
+    
+  }
+  latest_year <- max(wpp_years)
+  latest_pkg  <- paste0("wpp", latest_year)
+  
+  message("Latest available wpp package: ", latest_pkg)
+  # ---------------------------------------------------------------------------- #
+  
+  # ---------------------------------------------------------------------------- #
+  # Step 3: Check which wpp package (if any) is currently installed
+  installed_pkgs <- utils::installed.packages()[, "Package"]
+  installed_wpp  <- installed_pkgs[grepl("^wpp\\d{4}$", installed_pkgs)]
+  
+  if(length(installed_wpp) == 0) {
+    
+    message("No wpp package currently installed.")
+    
+    installed_version <- NA_character_
+    
+  } else {
+    
+    installed_wpp_years <- as.integer(sub("wpp", "", installed_wpp))
+    current_pkg         <- installed_wpp[which.max(installed_wpp_years)]
+    installed_version   <- as.character(packageVersion(current_pkg))
+    
+    message("Installed wpp package: ", current_pkg, " (v", installed_version, ")")
+  }
+  
+  # ---------------------------------------------------------------------------- #
+  
+  # ---------------------------------------------------------------------------- #
+  # Step 4: Get version of the latest package from GitHub
+  desc_url <- paste0("https://raw.githubusercontent.com/PPgp/", latest_pkg, "/main/DESCRIPTION")
+  
+  desc <- tryCatch(
+    httr::content(httr::GET(desc_url), "text", encoding = "UTF-8"),
+    error = function(e)
+      stop("Could not read DESCRIPTION for ", latest_pkg)
+  )
+  
+  latest_version <- sub("^Version:\\s+", 
+                        "",
+                        grep("^Version:", strsplit(desc, "\n")[[1]], value = TRUE))
+  # ---------------------------------------------------------------------------- #
+  
+  # ---------------------------------------------------------------------------- #
+  # Step 5: Compare versions
+  needs_update <- is.na(installed_version) || 
+    utils::compareVersion(installed_version, latest_version) < 0
+  
+  if(needs_update) {
+    
+    message("Newer version available: ", latest_pkg, " (v", latest_version, ")")
+    
+    ans <- readline(prompt = "Do you want to update to the latest version? [y/n]: ")
```

Comment:
check first if we're in an interactive session. If running on shiny, source, or batch, then this fails. use function interactive(). If FALSE, assume TRUE rather than breaking.